reviews/models/mixins.py

```diff
@@ -182,17 +192,66 @@ def resubmission_allowed(self, ev):
+            reviews_signals.reviews_email.send(context=context, caller='notify_edit_comment')
+
+    def get_accept_reject_notification(self):
+        context = self.get_context()
+        context['template'] = 'reviews_submission_status',
+        context['notify_comment'] = not self.reviewable.provider.reviews_comments_private and self.action.comment,
+        context['is_rejected'] = self.action.to_state == workflow.States.REJECTED.value,
+        reviews_signals.reviews_email.send(context=context, caller='accept_reject_notification')
+
+    def get_context(self):
+        return {
+            'settings': settings.DOMAIN,
+            'email_recipients': [contributor._id for contributor in self.reviewable.node.contributors],
+            'reviewable': self.reviewable,
+            'workflow': self.reviewable.provider.reviews_workflow,
+            'provider_url': self.reviewable.provider.external_url if self.reviewable.provider.external_url is not None else settings.DOMAIN + '/preprints' + self.reviewable.provider._id,
+            'provider_contact_email': self.reviewable.provider.email_contact if self.reviewable.provider.email_contact is not None else 'contact@osf.io',
+            'provider_support_email': self.reviewable.provider.email_support if self.reviewable.provider.email_support is not None else 'support@osf.io',
+        }
+
+@reviews_signals.reviews_email.connect
+def reviews_notification(self, context, caller):
+    timestamp = timezone.now()
+    event_type = utils.find_subscription_type('global_reviews')
+    template = ''.join(context.get('template')) + '.txt.mako'
+    for user_id in context.get('email_recipients'):
+        user = OSFUser.load(user_id)
+        subscriptions = get_user_subscriptions(user, event_type)
+        if user == context.get('reviewable').node.creator:
+            context['is_creator'] = True
+        else:
+            context['is_creator'] = False
+        for notification_type in subscriptions:
+            check_notify_comments = not (context['reviewable'].provider.reviews_comments_private and caller == 'notify_edit_comment')  # check if provider settings for comment notification
+            check_user_subscribe = subscriptions[notification_type] and user_id in subscriptions[notification_type]  # check if user is subscribed to this type of notifications
+            check_submission = caller == 'notify_submit' or notification_type != 'none'  # check if submission and bypass user subscription if none. Users will receive email for submission only with no more notifications in future.
+            if (check_submission and check_notify_comments and check_user_subscribe):
+                node_lineage_ids = get_node_lineage(context.get('reviewable').node) if context.get('reviewable').node else []
+                context['user'] = user
+                context['no_future_emails'] = caller == 'notify_submit' and notification_type == 'none'
```

Comment:
All the templates use `no_future_emails`, but this means it will only ever be true in `reviews_submission_confirmation`. Is that right? Can the other templates be simplified?

Or would it be better to just use `notification_type == 'none'`, in case we ever want to use it with other notifications?